server/controllers/api/assessmentDayDataController/assessmentDayDataController.test.js

```diff
@@ -307,59 +302,64 @@ describe('assessmentDayDataController', () => {
 
         await AssessmentDayDataController.create(request, response)
 
-        const assessmentVariables = await appDb
-          .collection(collections.assessmentVariables)
-          .find({ assessment: newAssessment }, { projection: { _id: 0 } })
-          .sort({ name: 1 })
-          .toArray()
         const assessmentDocument = await appDb
           .collection(collections.assessments)
           .findOne({
             name: newAssessment,
           })
+        const assessmentVariables = await appDb
+          .collection(collections.assessmentVariables)
+          .find(
+            { assessment_id: assessmentDocument._id },
+            { projection: { _id: 0, assessment_id: 0 } }
+          )
+          .sort({ name: 1 })
+          .toArray()
 
         expect(assessmentVariables).toEqual(assessment_variables)
         expect(assessmentDocument.name).toEqual(newAssessment)
       })
       it('Updates the assessment variables when there are new variables', async () => {
         const newAssessment = 'update-vars-assessment'
-        const intialVariables = initialAssessmentVariables(newAssessment)
+        const assessmentAttributes = AssessmentModel.withDefaults({
+          name: newAssessment,
+        })
+        const assessmentDoc = await AssessmentModel.upsert(
```

Comment:
this test example seems to be testing this model method instead of the controller action. To test the controller action we shouldn't have to upsert anything in the model.

Reply:
For this test my goal is to set it up and replicate the workflow, and I used the model functions to insert data in the database that best replicates what would happen in production.

Is the suggestion to insert the documents without the model..?